src/libsync/localdiscoverytracker.cpp

```diff
@@ -64,23 +64,48 @@ void LocalDiscoveryTracker::slotItemCompleted(const SyncFileItemPtr &item)
     //
     // For failures, we want to add the file to the list so the next sync
     // will be able to retry it.
-    if (item->_status == SyncFileItem::Success
-        || item->_status == SyncFileItem::FileIgnored
-        || item->_status == SyncFileItem::Restoration
-        || item->_status == SyncFileItem::Conflict
-        || (item->_status == SyncFileItem::NoStatus
-               && (item->_instruction == CSYNC_INSTRUCTION_NONE
-                      || item->_instruction == CSYNC_INSTRUCTION_UPDATE_METADATA))) {
-        if (_previousLocalDiscoveryPaths.erase(item->_file))
+
+    switch (item->_status) {
+    case SyncFileItem::NoStatus:
+        // we can't use the flags operator with CSYNC_INSTRUCTION_NONE
+        if (item->_instruction != CSYNC_INSTRUCTION_NONE && item->_instruction != CSYNC_INSTRUCTION_UPDATE_METADATA) {
+            break;
+        }
+        Q_FALLTHROUGH();
```

Comment:
This is clever, and _checkers_ of course know that macro. But not all people are Qt masters, and a comment what that macro does and how it changes the code flow would be great. As this changes the flow *significantly*.

Reply:
If I start do document the Qt functions I'll be getting no where.

Reply:
Hm, you miss my point here.

Reply:
It won't help with the readability but on master we can directly use:
https://en.cppreference.com/w/cpp/language/attributes/fallthrough